conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -214,8 +215,23 @@ def generate(self):
             else:
                 cache_variables[name] = value
 
+        # Get environment information for the presets
+        # FIXME: check if we have an alternative to avoid this way of getting this
+
+        prev_status = self._conanfile.virtualbuildenv
+        build_env = VirtualBuildEnv(self._conanfile).vars()
+        self._conanfile.virtualbuildenv = prev_status
+        buildenv = {name: value for name, value in
+                    build_env.items(variable_reference="$penv{{{name}}}")}
```

Comment:
Is this something to fix? This looks necessary, the placeholder really differs for different usage (cmd, powershell), and this is not known by VirtualBuildEnv until variable generation time.

Reply:
Sorry, no, just for the way of getting the environment, this is ok, I'll change the comment to clarify